controllers/spec/common.go

```diff
@@ -511,6 +516,10 @@ func getLegacyDownloadCommand(downloadPath, componentPackage string, authProvide
 func getDownloadCommand(downloadPath, componentPackage string, tlsProvided, authProvided bool, tlsConfig TLSConfig,
 	authConfig *v1alpha1.AuthConfig) []string {
 	var args []string
+	if hasHTTPPrefix(downloadPath) {
```

Comment:
Another question: any chance to make this feature backport to `getLegacyDownloadCommand`?

Reply:
looks like our runtime image neither have `wget` nor `curl`, so there will be some compatibility issues if add this feature to `getLegacyDownloadCommand`:

- we can add `wget` or `curl` to runtime images and add this feature
- but we also need to deprecate old runtime images or check whether given runtime images have `wget` or `curl` installed

I think there should be a general process for such a procedure(when we need to update runtime images):

- add `wget` or `curl` or other required utils to runtime images and release new versions
- add the new feature to FunctionMesh
- release FunctionMesh vx.y.z, and make an announcement that FunctionMesh vx.y.z is not compatible with runtime images older than va.b.c
- for user's custom runtime images, we list the requirements(what the image should contain)

HDYT? @freeznet @tpiperatgod @nlu90 ?

Reply:
This can be easily solved by Buildpacks, but we currently need a user-friendly image build workflow...

Reply:
are Buildpacks-related commits merged? or docs updated in functionmesh.io?

Reply:
has provided the docs and demos